### ISSUE TITLE
Fix S2583 FP: Variable Set in Catch Block

### ISSUE THREAD

**Issue by groogiam** [SonarSource/sonar-dotnet#8378]
### Description

S2583 FP Variable Set in Catch Block

### Repro steps

```chsarp
var success = true;
try
{
    var result = 100 / DateTime.Now.Second;
}
catch (Exception ex)
{
    success = false;
}
if (success) // FP
{
    Console.WriteLine("Success!");
}
```

### Expected behavior

S2583 should not trigger at success if statement.

### Actual behavior

S2583 is triggered at success if statement.

### Known workarounds

Please provide a description of any known workarounds.

### Related information

* C#/VB.NET Plugins version Sonarlint 7.4.0.80741
* Visual Studio version 17.8.0
* MSBuild / dotnet version 8.0.100
* SonarScanner for .NET version (if used)
* Operating System Windows 10

**Comment by Tim-Pohlmann** [comment]
Hi @groogiam thanks for the report!
I confirm this as an FP. I simplified the example a bit. The problem is, that the analyzer treats the division as a safe operation (which is wrong). We will improve this in the future.

**Comment by groogiam** [comment]
@Tim-Pohlmann Has this been released in Sonar Lint? I'm still seeing similar false positives to this in my code. Thanks.

**Comment by Tim-Pohlmann** [comment]
@groogiam Not yet. It should be in the next release. You can track the relevant issue [here](https://github.com/SonarSource/sonarlint-visualstudio/issues/5120).
Alternatively, you can download the newest analyzer (which contains the fix already) directly from [Nuget](https://www.nuget.org/packages/SonarAnalyzer.CSharp/).
